File: app/src/main/java/com/leo/nopasswordforyou/login_page.java
package com.leo.nopasswordforyou;

import static android.content.ContentValues.TAG;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;
import com.google.firebase.auth.FirebaseAuth;

import java.util.Objects;

public class login_page extends AppCompatActivity {

    FirebaseAuth mAuth = FirebaseAuth.getInstance();
    androidx.appcompat.app.AlertDialog alertDialog;
    TextView doSignUp, doSignIn;
    LinearLayout loginLayout, signUpLayout;
    //  private FirebaseFirestore db = FirebaseFirestore.getInstance();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login_page);


        mAuth.addAuthStateListener(firebaseAuth -> {
            mAuth.getCurrentUser();
            if (mAuth.getCurrentUser() != null) {
                startActivity(new Intent(this, ShowPass.class));
                mAuth.removeAuthStateListener((FirebaseAuth.AuthStateListener) this);
                finish();
            }

        });

        doSignUp = findViewById(R.id.doSignUp);
        doSignIn = findViewById(R.id.doSignIn);
        loginLayout = findViewById(R.id.loginLayout);
        signUpLayout = findViewById(R.id.signUpLayout);
        doSignUp.setOnClickListener(v -> {
            loginLayout.setVisibility(View.GONE);
            signUpLayout.setVisibility(View.VISIBLE);
        });
        doSignIn.setOnClickListener(v -> {
            loginLayout.setVisibility(View.VISIBLE);
            signUpLayout.setVisibility(View.GONE);
        });

        // log in
        EditText emailTextLogIn = findViewById(R.id.emailTextLogIn);
        EditText passTextLogIn = findViewById(R.id.passTextLogIn);
        Button logIn = findViewById(R.id.logIn);


        alertDialog = new MaterialAlertDialogBuilder(this).setView(R.layout.loading_dilogue_2).create();
        alertDialog.setCanceledOnTouchOutside(false);
        alertDialog.setCancelable(false);

        logIn.setOnClickListener(v -> {
            alertDialog.show();
            if (emailTextLogIn.getText().toString().trim().isEmpty() || passTextLogIn.getText().toString().trim().isEmpty()) {
                Toast.makeText(this, "Fields are empty", Toast.LENGTH_SHORT).show();
                //  makeToast(this,"Fields are empty");
                alertDialog.dismiss();
            } else {
                if (mAuth.getCurrentUser() != null) mAuth.signOut();

                loginAccount(emailTextLogIn.getText().toString().trim(), passTextLogIn.getText().toString().trim());
            }
        });

        //sign up
        EditText emailTextSignUp = findViewById(R.id.emailTextSignUp);
        EditText passTextSignUp = findViewById(R.id.passTextSignUp);
        Button btnSignUp = findViewById(R.id.btnSignUp);

        btnSignUp.setOnClickListener(v -> {
            alertDialog.show();
            if (emailTextSignUp.getText().toString().trim().isEmpty()
                    || passTextSignUp.getText().toString().trim().isEmpty()) {

                Toast.makeText(this, "fields are empty !", Toast.LENGTH_SHORT).show();
                alertDialog.dismiss();
            } else {
                if (mAuth.getCurrentUser() != null) {
                    mAuth.signOut();
                }
                createAccount(emailTextSignUp.getText().toString(), passTextSignUp.getText().toString());


            }
        });
    }




    private void loginAccount(String email, String password) {
        mAuth.signInWithEmailAndPassword(email, password)
                .addOnCompleteListener(this, task -> {
                    if (task.isSuccessful()) {
                        Objects.requireNonNull(mAuth.getCurrentUser()).reload();
                        // Sign in success, update UI with the signed-in user's information
                        alertDialog.dismiss();
                        //   loggedIn();

                        Log.d(TAG, "signInWithEmail:success");
                    }
                }).addOnFailureListener(e -> {
                    String error = e.getMessage();
                    Toast.makeText(this, error, Toast.LENGTH_SHORT).show();
                    alertDialog.dismiss();
                });
    }

    private void createAccount(String email, String password) {
        mAuth.createUserWithEmailAndPassword(email, password)
                .addOnCompleteListener(this, task -> {
                    alertDialog.dismiss();
                    if (task.isSuccessful()) {

                        // loggedIn();
                        Log.d(TAG, "createUserWithEmail:success");
                    } else {
                        Toast.makeText(this, Objects.requireNonNull(task.getException()).getMessage(), Toast.LENGTH_SHORT).show();
                        //makeToast(this, );
                    }
                }).addOnFailureListener(e -> {
                    alertDialog.dismiss();
                    Toast.makeText(this, e.getMessage(), Toast.LENGTH_SHORT).show();
                });
    }
}

File: app/src/main/java/com/leo/nopasswordforyou/activities/login_page.kt
/*
 *  No password for you
 *  Created by RudraOp9
 *  Modified on 25/02/24, 4:27 pm
 *  Copyright (c) 2024 . All rights reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package com.leo.nopasswordforyou.activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.google.firebase.auth.FirebaseAuth
import com.leo.nopasswordforyou.R
import com.leo.nopasswordforyou.databinding.ActivityLoginPageBinding
import com.leo.nopasswordforyou.viewmodel.Login_pageVM
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class login_page : AppCompatActivity() {
    var mAuth: FirebaseAuth = FirebaseAuth.getInstance()


    private var binding: ActivityLoginPageBinding? = null


    //  private FirebaseFirestore db = FirebaseFirestore.getInstance();
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (mAuth.currentUser != null) {
            startActivity(Intent(this@login_page, ShowPass::class.java))
            finish()
        }
        binding = ActivityLoginPageBinding.inflate(
            layoutInflater
        )
        val view: View = binding!!.root
        setContentView(view)
        val vm = ViewModelProvider(this)[Login_pageVM::class.java]
        vm.init(this, mAuth)
        val alertDialog =
            MaterialAlertDialogBuilder(this).setView(R.layout.loading_dilogue_2).create()
        alertDialog.setCanceledOnTouchOutside(false)
        alertDialog.setCancelable(false)





        binding!!.doSignUp.setOnClickListener {
            binding!!.loginLayout.visibility = View.GONE
            binding!!.signUpLayout.visibility = View.VISIBLE
        }
        binding!!.doSignIn.setOnClickListener {
            binding!!.loginLayout.visibility = View.VISIBLE
            binding!!.signUpLayout.visibility = View.GONE
        }


        // log in

        // log in
        binding!!.logIn.setOnClickListener {
            if (binding!!.emailTextLogIn.text.toString().trim { it <= ' ' }
                    .isEmpty() || binding!!.passTextLogIn.text.toString().trim { it <= ' ' }
                    .isEmpty()) {
                Toast.makeText(this, "Fields are empty", Toast.LENGTH_SHORT).show()
                //  makeToast(this,"Fields are empty");
            } else {
                if (mAuth.currentUser != null) mAuth.signOut()
                alertDialog.show()
                vm.loginAccount(
                    binding!!.emailTextLogIn.text.toString().trim { it <= ' ' },
                    binding!!.passTextLogIn.text.toString().trim { it <= ' ' },
                    this
                ) { s: String? ->
                    Toast.makeText(this, s, Toast.LENGTH_SHORT).show()
                    alertDialog.dismiss()
                    if (mAuth.currentUser != null) {
                        startActivity(Intent(this@login_page, ShowPass::class.java))
                        finish()
                    }
                }
            }
        }


        //sign up
        binding!!.btnSignUp.setOnClickListener {
            if (binding!!.emailTextSignUp.text.toString().trim { it <= ' ' }
                    .isEmpty()
                || binding!!.passTextSignUp.text.toString().trim { it <= ' ' }.isEmpty()
            ) {
                Toast.makeText(this, "fields are empty !", Toast.LENGTH_SHORT).show()
            } else {
                if (mAuth.currentUser != null) {
                    mAuth.signOut()
                }
                alertDialog.show()
                vm.createAccount(
                    binding!!.emailTextSignUp.text.toString(),
                    binding!!.passTextSignUp.text.toString(),
                    this
                ) { s: String? ->
                    alertDialog.dismiss()
                    Toast.makeText(this, s, Toast.LENGTH_SHORT).show()
                    if (mAuth.currentUser != null) {
                        startActivity(Intent(this@login_page, ShowPass::class.java))
                        finish()
                    }

                }
            }
        }
    }
}